### PR TITLE
Code QA automation for btest

### PR DESCRIPTION
**Pull request by ckreibich**
This applies the same toolchain that we recently added to `zkg`: black for formatting, pyupgrade for modernization, and flake8 for linting. I'm hoping we can get this in still before we release 1.0.0.

@timwoj I took out the sphinx entry in `.gitignore` — I think that was an oversight?

### CODE REVIEW

**Comment by ckreibich** [comment]
Looks like we have some unstable tests. I saw the `threads` test fail once  on macOS over in my fork, and `process` just failed once on Windows but passed upon re-run. Hrrm.

**Comment by ckreibich** [comment]
For tracking:
macOS failure: https://github.com/ckreibich/btest/actions/runs/4029678767/jobs/6927789749
Windows failure: https://github.com/zeek/btest/actions/runs/4029784697/jobs/6927982722

**Review by awelzel — APPROVED** [review]
I've put some comments around the f-stringification that I don't follow grasp the rules around, but don't think it's ~worse~ worth actually doing. Maybe as cleanups later.

I'd be okay doing this, but things looks different now. Also, some things certainly look worse now. Mainly calls with many arguments or funky string formatting, but that should straighten out writing a bit differently over time.

File: btest [hunk 1]
```diff
@@ -198,32 +207,30 @@ def readStateFile():
...
-                                     encoding='utf-8',
-                                     suffix='.sh',
-                                     dir=tmpdir,
-                                     delete=True)
-    fcontents = '#!/usr/bin/env bash\n%s\n' % cmd
+    """
+    tf = tempfile.NamedTemporaryFile(
+        mode="w", encoding="utf-8", suffix=".sh", dir=tmpdir, delete=True
+    )
+    fcontents = "#!/usr/bin/env bash\n%s\n" % cmd
```
[the remark this answers is not in the document]
Comment (ckreibich) [reply btest:229]: I played with this a bit. Looks like it only goes near the %-substitution when the argument is a tuple. With that, it next upgrades to `.format()`, and on another pass it puts in the f-string.

File: btest [hunk 2]
```diff
@@ -937,13 +966,15 @@ class Test(object):
...
 
         # Make sure all non-optional commands are there.
-        for (tag, regexp, multiple, optional, group1, group2) in Commands:
+        for tag, regexp, multiple, optional, group1, group2 in Commands:
             if not optional and tag not in cmds:
                 if tag == "exec":
-                    error("%s: mandatory keyword '@TEST-EXEC' or '@TEST-EXEC-FAIL' is missing." %
-                          file)
+                    error(
+                        "%s: mandatory keyword '@TEST-EXEC' or '@TEST-EXEC-FAIL' is missing."
```
[the remark this answers is not in the document]
Comment (ckreibich) [reply btest:973]: Yeah. When I make it `% (file,)` (i.e., also put it on the same line), it becomes an f-string, but black always seems to find it too long.

File: btest [hunk 3]
```diff
@@ -1388,24 +1450,32 @@ class Test(object):
 
         cmdline = RE_DIR.sub(self.dir, cmdline)
 
-        print("%s (expect %s)" % (cmdline, ("failure", "success")[cmd.expect_success]),
-              file=self.log)
+        print(
```
[the remark this answers is not in the document]
Comment (ckreibich) [reply btest:1451]: Done, yep. Thanks.

**Comment by ckreibich** [comment]
@rsmmr does this all look okay to you? Thanks...

**Review by rsmmr — APPROVED** [review]
Yeah, this looks fine to me. 

The one `black` thing I don't like is the frequent break-after-open-parenthesis for long lines, in particular for `if`-statements, like:

```
if (
    Options.abort_on_failure
    ...
    )
```

But I can live with that if that's the style we have settled on.

**Comment by ckreibich** [comment]
I agree with this — not perfect, but overall pretty nice. It's in place now in `zkg` too. Thanks.

**Pull request merged**